Explorer/Assets/Scripts/Global/Dynamic/RealmController.cs

```diff
@@ -86,9 +90,44 @@ public async UniTask SetRealmAsync(URLDomain realm, CancellationToken ct)
             if (!ComplimentWithStaticPointers(world, realmEntity) && !realmComp.ScenesAreFixed)
                 ComplimentWithVolatilePointers(world, realmEntity);
 
-            teleportController.OnRealmLoaded(realmData.ScenesAreFixed ? retrieveSceneFromFixedRealm : retrieveSceneFromVolatileWorld);
+            IRetrieveScene sceneProviderStrategy = realmData.ScenesAreFixed ? retrieveSceneFromFixedRealm : retrieveSceneFromVolatileWorld;
+            sceneProviderStrategy.World = globalWorld.EcsWorld;
+            teleportController.SceneProviderStrategy = sceneProviderStrategy;
 
-            // Hide loading screen
+            var sceneLoadReport = new AsyncLoadProcessReport(new UniTaskCompletionSource(), new AsyncReactiveProperty<float>(0));
+
+            if (loadReport != null)
+                PropagateSceneLoadReportToRealmLoadReportAsync(loadReport.ProgressCounter.Value, loadReport, sceneLoadReport, ct).Forget();
+
+            try
+            {
+                await teleportController.TeleportToSceneSpawnPointAsync(playerStartPosition, sceneLoadReport, ct);
+
+                if (loadReport != null)
+                {
+                    loadReport.ProgressCounter.Value = 1f;
+                    loadReport.CompletionSource.TrySetResult();
+                }
+            }
+            catch (Exception e) { loadReport?.CompletionSource.TrySetException(e); }
+        }
+
+        private async UniTaskVoid PropagateSceneLoadReportToRealmLoadReportAsync(
```

Comment:
Let's move it to an utility class as it does not belong to `RealmController` directly